docs/advanced/signal_generators.md

```diff
@@ -0,0 +1,558 @@
+
+[^TOB]: Amplitude upper bound is reduced to 16 when the frequency is greater than 60MHz. It is further reduced to 12 when the frequency is greater than 80 MHz
+
+[^TOC]: Amplitude upper bound is reduced to 8 when the frequency is greater than 200MHz.
+
+```
+
+______________________________________________________________________
+
+## Arbitrary Waveform Generators
+
+```{autoclasstree} tm_devices.drivers.pi.signal_generators.awgs
+---
+full:
+namespace: tm_devices.drivers.pi.signal_generators.awgs
+align: center
+alt: Complete device driver class diagram
+---
+```
+
+Arbitrary Waveform Generators require several different parameters to be specified for a waveform to be generated.
+
+Function generation on AWGs are fundamentally different from AFGs. The AWG is stopped and the channel being changed
+is turned off. Predefined waveforms provided with the AWG
+are then loaded from the hard drive into the waveform list for the AWG5200 and AWG70K. Sample rate is not channel dependant,
+instead being set through the source class. The channel provided has its waveform, offset, amplitude and signal path set.
+These attributes can take a while to be set, though once complete, the channels are turned back on and the `AWGCONTROL:RUN`
+is set.
+
+```{note}
+If the waveform is `RAMP`, a symmetry of 50 will set the waveform to a `TRIANGLE`.
+```
+
+The AWG class has some methods specific to it. `generate_waveform` allows for a waveform name from the waveform list
+to be provided, instead of a function. The method is also distinctly different from generate function as it relies on a sample
+rate also being provided to actually generate the waveform.
+
+AWGs have access to the following functions:
+`SIN`, `SQUARE`, `RAMP`, `TRIANGLE`, `DC`, `CLOCK`
+
+### AWG5K/AWG7K
+
+The AWG5K/7K series are signal sources focused on waveform generation which operate on Windows XP (Windows 7 for model C).
+They accept communication through LAN and GPIB interfaces.
+
+`set_output_signal_path` is uniquely defined within the AWG5K and AWG7K classes, as it will set the value for
+`AWGCONTROL:DOUTPUTx:STATE`, which is a unique option not seen in the other AWGs.
+
+`set_offset` is conditioned to make sure that the AWG output signal path is not DIR, as the VISA query will time
+out otherwise.
+
+```{note}
+Operation complete commands will always return 1 on the AWG5K/7K series.
+```
+
+```{caution}
+All waveforms must be of the same length on requesting AWGCONTROL:RUN.
+```
+
+#### AWG5K, AWG5KB, AWG5KC
+
+#### Constraints:
+
+The AWG5K series offers a upper sample rate range from 600 MS/s to 1.2 GS/s dependent on the model number. The amplitude range is dependent on
+the signal output path, and if the direct option is selected, it will reduce the amplitude range to 20mV to 0.6 V and the
+offset to 0. Otherwise, the amplitude ranges from 20 mV to 2.25 V and an offset range of plus or minus 2.25 V.
```

Comment:
Isn't the last sentence just duplicating what is in the table?

Reply:
It can be removed.